### ISSUE TITLE
MemoryMappedFile CreateFromFileTestCases failed in CI

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#1061]
```
Discovering: MemoryMappedFile.Tests
  Discovered:  MemoryMappedFile.Tests
  Starting:    MemoryMappedFile.Tests
  ERR999: Unexpected exception in runTest, System.IO.IOException: The process cannot access the file 'd:\j\workspace\dotnet_corefx_windows_debug_prtest\bin\tests\Windows_NT.AnyCPU.Debug\MemoryMappedFile.Tests\aspnetcore50\CreateFromFile_test2_5f7a364a-e0be-4a98-abcd-67568018dfa7.txt' because it is being used by another process.
     at System.IO.Win32FileStream.Init(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs)
     at System.IO.Win32FileSystem.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
     at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
     at System.IO.FileStream..ctor(String path, FileMode mode)
     at CreateFromFile.RunTest() in d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.IO.MemoryMappedFiles\tests\MemoryMappedFile\CreateFromFile.cs:line 385
     CreateFromFile.CreateFromFileTestCases [FAIL]
        One or more test cases failed.
        Stack Trace:
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.IO.MemoryMappedFiles\tests\MemoryMappedFile\CreateFromFile.cs(34,0): at CreateFromFile.CreateFromFileTestCases()
  Finished:    MemoryMappedFile.Tests
  
  === TEST EXECUTION SUMMARY ===
     MemoryMappedFile.Tests  Total: 7, Errors: 0, Failed: 1, Skipped: 0, Time: 1.641s
```

**Comment by joshfree** [comment]
@venkat-raman251 could you investigate?

**Comment by stephentoub** [comment]
Fixed by #2334
